### PR TITLE
MAINT: Remove Python2 workarounds in distutils

### PR DESCRIPTION
**Pull request by sethtroisi**
More #15306 work

### CODE REVIEW

File: numpy/distutils/misc_util.py [hunk 1]
```diff
@@ -318,6 +318,7 @@ def make_temp_file(suffix='', prefix='', text=True):
 # Hooks for colored terminal output.
 # See also https://web.archive.org/web/20100314204946/http://www.livinglogic.de/Python/ansistyle
 def terminal_has_colors():
+    # Can this be cleaned up?
```
Comment (sethtroisi) [inline numpy/distutils/misc_util.py]: this code goes back to at least 2005, I suspect it's fine now but I don't have a way to check.

File: numpy/distutils/mingw32ccompiler.py [hunk 2]
```diff
@@ -141,17 +146,16 @@ def __init__ (self,
...
             else:
                 # gcc-4 series releases do not support -mno-cygwin option
                 self.set_executables(compiler='gcc -O2 -Wall',
                                      compiler_so='gcc -O2 -Wall -Wstrict-prototypes',
+                                     compiler_cxx='g++',
                                      linker_exe='g++ ',
                                      linker_so='g++ -shared')
-        # added for python2.3 support
-        # we can't pass it through set_executables because pre 2.2 would fail
-        self.compiler_cxx = ['g++']
```
Comment (sethtroisi) [inline numpy/distutils/mingw32ccompiler.py]: It looks like this is maybe still not supported by MSVCCompiler

`AttributeError: 'MSVCCompiler' object has no attribute 'ldflags_shared'`

**Comment by mattip** [comment]
Without a working mingw CI run or a local run with both MSVC and mingw, I would be very reluctant to merge this. Perhaps better to leave this as is until someone with mingw can confirm that it does not break working builds.

**Comment by sethtroisi** [comment]
I'll close till I can give this a 2nd look.

**Pull request closed without merging**